src/Bijectors.jl

```diff
@@ -84,213 +109,163 @@ function eachcolmaphcat(f, x)
-    elseif lowerbounded
-        return log(x - a)
-    elseif upperbounded
-        return log(b - x)
+function logpdf_with_trans(d::UnivariateDistribution, x::AbstractArray{<:Real}, trans::Bool)
+    if toflatten(d)
+        f, args = flatten(d, trans)
+        return f.(args..., x)
     else
-        return x
+        return map(x) do x
+            logpdf_with_trans(d, x, trans)
+        end
     end
 end
 
-invlink(d::TransformDistribution, y::Real) = _clamp(_invlink(d, y), d)
-function _invlink(d::TransformDistribution, y::Real)
-    a, b = minimum(d), maximum(d)
-    lowerbounded, upperbounded = isfinite(a), isfinite(b)
-    if lowerbounded && upperbounded
-        return (b - a) * StatsFuns.logistic(y) + a
-    elseif lowerbounded
-        return exp(y) + a
-    elseif upperbounded
-        return b - exp(y)
-    else
-        return y
-    end
+function logpdf_with_trans(d::DiscreteUnivariateDistribution, x::Integer, transform::Bool)
+    return logpdf(d, x)
 end
 
+## Multivariate
+
 function logpdf_with_trans(
-    d::TransformDistribution,
-    x::Real,
-    transform::Bool,
+    dist::Distributions.Product,
+    x::AbstractVector{<:Real},
+    istrans::Bool,
 )
-    return _logpdf_with_trans(d, x, transform)
+    return sum(maporbroadcast(dist.v, x) do d, x
+        logpdf_with_trans(d, x, istrans)
+    end)
 end
 function logpdf_with_trans(
-    d::TransformDistribution,
-    x::AbstractArray{<:Real},
-    transform::Bool,
+    dist::Distributions.Product,
+    x::AbstractMatrix{<:Real},
+    istrans::Bool,
 )
-    return mapvcat(x -> _logpdf_with_trans(d, x, transform), x)
-end
-function _logpdf_with_trans(d::TransformDistribution, x::Real, transform::Bool)
-    lp = logpdf(d, x)
-    if transform
-        x = _clamp(x, d)
-        a, b = minimum(d), maximum(d)
-        lowerbounded, upperbounded = isfinite(a), isfinite(b)
-        if lowerbounded && upperbounded
-            lp += log((x - a) * (b - x) / (b - a))
-        elseif lowerbounded
-            lp += log(x - a)
-        elseif upperbounded
-            lp += log(b - x)
-        end
+    return map(eachcol(x)) do x
+        logpdf_with_trans(dist, x, istrans)
     end
-    return lp
 end
 
-#########
-# 0 < x #
-#########
-
-const PositiveDistribution = Union{
-    BetaPrime, Chi, Chisq, Erlang, Exponential, FDist, Frechet, Gamma, InverseGamma,
-    InverseGaussian, Kolmogorov, LogNormal, NoncentralChisq, NoncentralF, Rayleigh, Weibull,
-}
-
-_link(d::PositiveDistribution, x::Real) = log(x)
-_invlink(d::PositiveDistribution, y::Real) = exp(y)
-function _logpdf_with_trans(d::PositiveDistribution, x::Real, transform::Bool)
-    return logpdf(d, x) + transform * log(x)
+function link(
+    dist::Distributions.Product,
+    x::AbstractVector{<:Real},
+)
+    return maporbroadcast(link, dist.v, x)
+end
+function link(
+    dist::Distributions.Product,
+    x::AbstractMatrix{<:Real},
+)
+    return map(eachcol(x)) do x
+        link(dist, x)
+    end
 end
 
+function invlink(
```

Comment:
These impls shouldn't be necessary, but instead be handled by the batch-version for `Stacked`, no?

Reply:
I am making these distributions' bijectors the `DistributionBijector` simply because I was too lazy to define a new bijector type but that's the correct approach.  See the above comment for details.